<!DOCTYPE html>
<html lang="Zh-cn">

  <head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>bamdst使用 | Liripo</title>
    <link rel="shortcut icon" href="../../image/title.svg" type="image/x-icon">
  <script src="../../fontawesome-free-5.11.2-web/fontawesome-free-5.11.2-web/js/all.js" crossorigin="anonymous"></script>

  <style>
  	:root {
  		--pagecolour: #333333;
  		--maintext: #dddddd;
  		--faded: #888888;
  		--highlight: #ff69b4;
  		--lowlight: #ba68c8;
	}
  </style>

  <link rel="stylesheet" href="../../css/et_book.css" />
  <link rel="stylesheet" href="../../css/style.css" />

  
<meta name="twitter:card" content="summary">
<meta property="og:title" content="bamdst使用">




</head>


  <body>
    



    <div class="main-content">
      <nav>
  <ul class="menu">
    
    <li><a href="../../">Home</a></li>
    
    <li><a href="../../about/">About</a></li>
    
    <li><a href="../../post/">Posts</a></li>
    
    <li><a href="../../tags/">Tags</a></li>
    
  </ul>
</nav>

      <div>
  <h1>bamdst使用</h1>
  <h3>
    17 Dec 2019
  </h3>
   <h5 id="wc">1225 Words|Read in about 3       Min|本文总阅读量<span id="busuanzi_value_page_pv"></span>次</h5>
    </h5>
  <br><br>
</div>


      <main>
        

        <p>bamdst:用于计算bam文件测序深度和覆盖度（Sequencing depth and coverage）,有个疑问，应该使用哪个bam呢？？使用bwa及samtools得到的bam测序深度很高。。</p>
<h3 id="heading">安装</h3>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">git clone https://github.com/shiquan/bamdst.git ~
cd bamdst/
make
</code></pre></div><p>#安装完后会在项目根目录生成一个bamdst程序</p>
<pre><code>-rwxrwxr-x 1 liaorp liaorp 532816 Dec 17 11:02 bamdst
</code></pre><p>可以看到文件权限可执行,键入<code>./bamdst</code>即可运行，或者加入环境变量中。</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">echo -e <span style="color:#e6db74">&#39;#添加bamdst\nexport PATH=~/software/bamdst:$PATH&#39;</span>&gt;&gt;~/.bashrc#记住是单引号
source ~/.bashrc#使环境生效
</code></pre></div><h3 id="heading-1">概念</h3>
<p><strong>测序深度Depth</strong>：测序得到的总碱基数与待测基因组大小的比值。如人的基因组为3Gb，测序获得90Gb数据量，平均测序深度为30X。
<strong>覆盖率</strong>：</p>
<h3 id="example">example</h3>
<p>下载数据有个example文件夹，测试以下,使用-p,-o参数</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">cd example <span style="color:#f92672">&amp;&amp;</span> mkdir test
bamdst -p MT-RNR1.bed -o ./test test.bsm
</code></pre></div><p>产生七个文件:</p>
<pre><code>$ls
chromosomes.report  coverage.report  depth_distribution.plot  depth.tsv.gz  insertsize.plot  region.tsv.gz  uncover.bed
</code></pre><ul>
<li>chromosomes.report:该文件中存储的是从bam文件中获取的目标染色体深度、覆盖度信息
bed文件即是为了挑选目标区域，即感兴趣的位置。</li>
<li>coverage.report：信息很多，只看这个文件即可
文件分成三部分[total],[Target],[Flank],其中文件中还有一个关键字rmdup</li>
</ul>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash"><span style="color:#f92672">[</span>Target<span style="color:#f92672">]</span> Average depth    0.26 
<span style="color:#f92672">[</span>Target<span style="color:#f92672">]</span> Average depth<span style="color:#f92672">(</span>rmdup<span style="color:#f92672">)</span>    0.06
<span style="color:#f92672">[</span>Target<span style="color:#f92672">]</span> Coverage <span style="color:#f92672">(</span>&gt;0x<span style="color:#f92672">)</span>    5.66% <span style="color:#75715e">#目标区域大于0X覆盖率，排除0X</span>
<span style="color:#f92672">[</span>Target<span style="color:#f92672">]</span> Coverage <span style="color:#f92672">(</span>&gt;<span style="color:#f92672">=</span>4x<span style="color:#f92672">)</span>    2.83%
<span style="color:#f92672">[</span>Target<span style="color:#f92672">]</span> Coverage <span style="color:#f92672">(</span>&gt;<span style="color:#f92672">=</span>10x<span style="color:#f92672">)</span>    0.00%
<span style="color:#f92672">[</span>Target<span style="color:#f92672">]</span> Coverage <span style="color:#f92672">(</span>&gt;<span style="color:#f92672">=</span>30x<span style="color:#f92672">)</span>    0.00%
<span style="color:#f92672">[</span>Target<span style="color:#f92672">]</span> Coverage <span style="color:#f92672">(</span>&gt;<span style="color:#f92672">=</span>100x<span style="color:#f92672">)</span>    0.00%
</code></pre></div><p>目标区域平均测序深度，算一下呗,即LN/G，where L is the read length, N is the number of reads and G is the haploid genome length.<sup id="fnref:1"><a href="#fn:1" class="footnote-ref" role="doc-noteref">1</a></sup>
可以使用<a href="http://www.chinapubmed.net/">中文pubmed</a></p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">$ awk <span style="color:#e6db74">&#39;{a=$1*$2;sum+=a;b=sum/954}END{print b}&#39;</span> depth_distribution.plot
0.261006
</code></pre></div><p>目标区域覆盖率：为文件depth_distribution.plot的第五列，1-第三列
在R中输入<code>round(54/954,6)</code>得到0.056604</p>
<ul>
<li>
<p>insertsize.plot：由于example得到的文件为空，有待探究</p>
</li>
<li>
<p>depth.tsv.gz：记录了每个位点的 Raw Depth       Rmdup depth     Cover depth，三个名词各自代表？列出前几列</p>
</li>
</ul>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash"><span style="color:#75715e">#Chr    Pos     Raw Depth       Rmdup depth     Cover depth</span>
chrM    <span style="color:#ae81ff">650</span>     <span style="color:#ae81ff">8</span>       <span style="color:#ae81ff">6</span>       8
chrM    <span style="color:#ae81ff">651</span>     <span style="color:#ae81ff">8</span>       <span style="color:#ae81ff">6</span>       8
chrM    <span style="color:#ae81ff">652</span>     <span style="color:#ae81ff">8</span>       <span style="color:#ae81ff">6</span>       8
chrM    <span style="color:#ae81ff">653</span>     <span style="color:#ae81ff">9</span>       <span style="color:#ae81ff">6</span>       9
chrM    <span style="color:#ae81ff">654</span>     <span style="color:#ae81ff">9</span>       <span style="color:#ae81ff">6</span>       9
chrM    <span style="color:#ae81ff">655</span>     <span style="color:#ae81ff">9</span>       <span style="color:#ae81ff">6</span>       9
chrM    <span style="color:#ae81ff">656</span>     <span style="color:#ae81ff">9</span>       <span style="color:#ae81ff">6</span>       9
chrM    <span style="color:#ae81ff">657</span>     <span style="color:#ae81ff">9</span>       <span style="color:#ae81ff">6</span>       9
chrM    <span style="color:#ae81ff">658</span>     <span style="color:#ae81ff">9</span>       <span style="color:#ae81ff">6</span>       9
</code></pre></div><ul>
<li>
<p>region.tsv.gz：记录bed文件每个区域的测序深度中位数覆盖率以及Coverage(FIX)？？，由于example的bed只有一个区域，所以可以直接去除以那个区域长度得到目标区域测序深度。</p>
</li>
<li>
<p>depth_distribution.plot :深度分布图，可以结合R绘图,如下，测序深度为9的有6个位点</p>
</li>
</ul>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash"><span style="color:#ae81ff">0</span>       <span style="color:#ae81ff">900</span>     0.943396        <span style="color:#ae81ff">54</span>      0.056604
<span style="color:#ae81ff">1</span>       <span style="color:#ae81ff">0</span>       0.000000        <span style="color:#ae81ff">54</span>      0.056604
<span style="color:#ae81ff">2</span>       <span style="color:#ae81ff">0</span>       0.000000        <span style="color:#ae81ff">54</span>      0.056604
<span style="color:#ae81ff">3</span>       <span style="color:#ae81ff">27</span>      0.028302        <span style="color:#ae81ff">27</span>      0.028302
<span style="color:#ae81ff">4</span>       <span style="color:#ae81ff">4</span>       0.004193        <span style="color:#ae81ff">23</span>      0.024109
<span style="color:#ae81ff">5</span>       <span style="color:#ae81ff">12</span>      0.012579        <span style="color:#ae81ff">11</span>      0.011530
<span style="color:#ae81ff">6</span>       <span style="color:#ae81ff">1</span>       0.001048        <span style="color:#ae81ff">10</span>      0.010482
<span style="color:#ae81ff">7</span>       <span style="color:#ae81ff">0</span>       0.000000        <span style="color:#ae81ff">10</span>      0.010482
<span style="color:#ae81ff">8</span>       <span style="color:#ae81ff">4</span>       0.004193        <span style="color:#ae81ff">6</span>       0.006289
<span style="color:#ae81ff">9</span>       <span style="color:#ae81ff">6</span>       0.006289        <span style="color:#ae81ff">0</span>       0.000000
</code></pre></div><p>算下理解是否正确</p>
<pre><code>$ awk '{sum+=$2}END{print sum}' depth_distribution.plot
954
$ expr 1603 - 649
954
</code></pre><ul>
<li>uncover.bed :没有捕获区域</li>
</ul>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">$ cat example/MT-RNR1.bed
chrM    <span style="color:#ae81ff">649</span>     1603
$ cat uncover.bed
chrM    <span style="color:#ae81ff">672</span>     1603
</code></pre></div><h3 id="heading-2">可选参数</h3>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash"><span style="color:#75715e">#方括号中为程序默认的参数值</span>
   -f, --flank <span style="color:#f92672">[</span>200<span style="color:#f92672">]</span>   flank n bp of each region
   -q <span style="color:#f92672">[</span>20<span style="color:#f92672">]</span>             map quality cutoff value, greater or equal to the value will be count
   --maxdepth <span style="color:#f92672">[</span>0<span style="color:#f92672">]</span>      set the max depth to stat the cumu distribution.
   --cutoffdepth <span style="color:#f92672">[</span>0<span style="color:#f92672">]</span>   list the coverage of above depths
   --isize <span style="color:#f92672">[</span>2000<span style="color:#f92672">]</span>      stat the inferred insert size under this value
   --uncover <span style="color:#f92672">[</span>5<span style="color:#f92672">]</span>       region will included in uncover file <span style="color:#66d9ef">if</span> below it
   --bamout  BAMFILE   target reads will be exported to this bam file
   -1                  begin position of bed file is 1-based
   -h, --help          print this help info
</code></pre></div><section class="footnotes" role="doc-endnotes">
<hr>
<ol>
<li id="fn:1" role="doc-endnote">
<p>Sims, D., Sudbery, I., Ilott, N. et al. Sequencing depth and coverage: key considerations in genomic analyses. Nat Rev Genet 15, 121–132 (2014) doi:10.1038/nrg3642<a href="#fnref:1" class="footnote-backref" role="doc-backlink">&#8617;</a></p>
</li>
</ol>
</section>
      </main>

      <footer>
  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.2/highlight.min.js"></script>
  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.2/languages/r.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.2/languages/python.min.js"></script>

  <script>
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  </script>
  
  <script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
  <span id="busuanzi_container_site_pv">本站总访问量<span id       ="busuanzi_value_site_pv"></span>次</span>
  <span id="busuanzi_container_site_uv">
    本站访客数<span id="busuanzi_value_site_uv"></span>人次
</span>
  
  
<script async src="../../js/center-image.js"></script>
<script type="text/javascript" id="MathJax-script" async
  src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js">
</script>
<center>
Liripo <SPAN id=span_dt_dt style="color: #0196e3;"></SPAN>
<SCRIPT language=javascript>
function show_date_time(){
window.setTimeout("show_date_time()", 1000);
BirthDay=new Date("7/8/2019 00:00:00");
today=new Date();
timeold=(today.getTime()-BirthDay.getTime());
sectimeold=timeold/1000
secondsold=Math.floor(sectimeold);
msPerDay=24*60*60*1000
e_daysold=timeold/msPerDay
daysold=Math.floor(e_daysold);
e_hrsold=(e_daysold-daysold)*24;
hrsold=Math.floor(e_hrsold);
e_minsold=(e_hrsold-hrsold)*60;
minsold=Math.floor((e_hrsold-hrsold)*60);
seconds=Math.floor((e_minsold-minsold)*60);
span_dt_dt.innerHTML=""+daysold+"天"+hrsold+"小时"+minsold+"分"+seconds+"秒";
}
show_date_time();
</SCRIPT>
</center>
  
</footer>


    </div>
  </body>
</html>
