<!DOCTYPE html>
<html lang="Zh-cn">
<head>
    <meta charset="utf-8">
    <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://liripo.github.io/css/frameworks.css" />
    <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://liripo.github.io/css/github.css" />
    
    <link crossorigin="anonymous" media="(prefers-color-scheme: dark)" rel="stylesheet"
        href="https://liripo.github.io/css/dark.css" />
    
    <meta name="viewport" content="width=device-width">
    <meta name="description"
        content=" 原始测序文件fq.gz经genome比对，变异检测得到vcf文件，亦即变异位点文件， 只是通过看vcf文件我们是不知道些变异位点到底是位于基因的exon、intron、UTR等的哪些区域的 ，所以需要注释， 常用的vcf注释软件有annovar和snpEff 。
" />
    <meta name="keywords"
        content="" />
    <meta name="robots" content="noodp" />
    <link rel="canonical" href="https://liripo.github.io/posts/2019/vcf%E6%96%87%E4%BB%B6%E6%B3%A8%E9%87%8A/" />

    
    <meta name="twitter:card" content="summary" />
    <meta name="twitter:title" content="vcf文件注释 - Liripo" />
    <meta name="twitter:description"
        content=" 原始测序文件fq.gz经genome比对，变异检测得到vcf文件，亦即变异位点文件， 只是通过看vcf文件我们是不知道些变异位点到底是位于基因的exon、intron、UTR等的哪些区域的 ，所以需要注释， 常用的vcf注释软件有annovar和snpEff 。
" />
    <meta name="twitter:site" content="https://liripo.github.io" />
    <meta name="twitter:creator" content="" />
    <meta name="twitter:image"
        content="https://liripo.github.io">

    
    <meta property="og:type" content="article" />
    <meta property="og:title" content="vcf文件注释 - Liripo">
    <meta property="og:description"
        content=" 原始测序文件fq.gz经genome比对，变异检测得到vcf文件，亦即变异位点文件， 只是通过看vcf文件我们是不知道些变异位点到底是位于基因的exon、intron、UTR等的哪些区域的 ，所以需要注释， 常用的vcf注释软件有annovar和snpEff 。
" />
    <meta property="og:url" content="https://liripo.github.io/posts/2019/vcf%E6%96%87%E4%BB%B6%E6%B3%A8%E9%87%8A/" />
    <meta property="og:site_name" content="vcf文件注释" />
    <meta property="og:image"
        content="https://liripo.github.io">
    <meta property="og:image:width" content="2048">
    <meta property="og:image:height" content="1024">
    
    <meta property="article:published_time" content="2020-01-20 23:58:40 &#43;0000 UTC" />
    

    
    

    
    

    <title>vcf文件注释 - Liripo</title>

    <link rel="icon" type="image/x-icon" class="js-site-favicon" href="https://liripo.github.io/images/favicon.ico">
    <meta name="theme-color" content="#1e2327">

    
    

    
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-155993181-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>

</head>

<body class="env-production emoji-size-boost page-responsive page-profile">
  <div class="position-relative js-header-wrapper ">
    <span class="Progress progress-pjax-loader position-fixed width-full js-pjax-loader-bar">
        <span class="progress-pjax-loader-bar top-0 left-0" style="width: 0%;"></span>
    </span>
    <header class="Header py-lg-0 js-details-container Details flex-wrap flex-lg-nowrap p-responsive" role="banner">
        <div class="Header-item d-none d-lg-flex">
            <a class="Header-link" href="https://liripo.github.io" aria-label="Homepage"
                data-ga-click="Header, go to dashboard, icon:logo">
                <svg class="octicon octicon-mark-github v-align-middle" height="32" viewBox="0 0 16 16" version="1.1"
                    width="32" aria-hidden="true">
                    <path fill-rule="evenodd"
                        d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z" />
                </svg>
            </a>
        </div>
        <div
            class="Header-item Header-item--full flex-column flex-lg-row width-full flex-order-2 flex-lg-order-none mr-0 mr-lg-3 mt-3 mt-lg-0 Details-content--hidden">
            <div class="header-search flex-self-stretch flex-lg-self-auto mr-0 mr-lg-3 mb-3 mb-lg-0 scoped-search site-scoped-search js-site-search position-relative js-jump-to"
                role="combobox" aria-owns="jump-to-results" aria-label="Search or jump to" aria-haspopup="listbox"
                aria-expanded="false">
                <div class="position-relative">
                </div>
            </div>
        </div>
        <nav class="d-flex" aria-label="Global">
            <a class="js-selected-navigation-item Header-link py-lg-3">&nbsp;</a>
        </nav>
        <div class="Header-item Header-item--full flex-justify-center d-lg-none position-relative" style="margin-right: auto;">
            <a class="Header-link" href="https://liripo.github.io" aria-label="Homepage"
                data-ga-click="Header, go to dashboard, icon:logo">
                <svg class="octicon octicon-mark-github v-align-middle" height="32" viewBox="0 0 16 16" version="1.1"
                    width="32" aria-hidden="true">
                    <path fill-rule="evenodd"
                        d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z" />
                </svg>
            </a>
        </div>
        <div class="Header-item position-relative mr-0 d-none d-lg-flex">
            <details class="details-overlay details-reset">
                <summary class="Header-link" aria-label="View profile and more"
                    data-ga-click="Header, show menu, icon:avatar">
                    
                    <img alt="" class="avatar" src="https://themes.gohugo.io/theme/github-style/images/avatar.png" height="20" width="20">
                    
                </summary>
            </details>
        </div>
    </header>
</div>

<div id="start-of-content" class="show-on-focus"></div>
<div id="js-flash-container">
</div>

  
<div class="application-main " data-commit-hovercards-enabled="">
    <div itemscope="" itemtype="http://schema.org/SoftwareSourceCode" class="">
        <main>
            <div class="pagehead repohead instapaper_ignore readability-menu experiment-repo-nav pt-0 pt-lg-4 ">
                <div class="repohead-details-container clearfix container-lg p-responsive d-none d-lg-block">
                    <div class="mb-3 d-flex">
                        <h1 class="public css-truncate float-none flex-auto width-fit pl-0">
                            <a class="avatar mr-1" href="https://liripo.github.io/about/">
                                

                                <img src="https://themes.gohugo.io/theme/github-style/images/avatar.png" width="26" height="26">
                                
                            </a>
                            <span class="author"><a href="https://liripo.github.io">liripo</a></span>
                            <span class="path-divider">/</span>
                            <strong itemprop="name" class="css-truncate-target" style="max-width: 410px"><a
                                    href="https://liripo.github.io/posts/2019/vcf%E6%96%87%E4%BB%B6%E6%B3%A8%E9%87%8A/">vcf文件注释</a></strong>

                            <div class="d-block text-small text-gray">
                                Created <time-ago datetime="2020-01-20" class="no-wrap"
                                    title="Created at 2020/01/20">
                                    2020-01-20</time-ago>
                                <span class="file-info-divider"></span>
                                Modified <time-ago datetime="2020-01-20" class="no-wrap"
                                    title="Modified  at 2020/01/20">
                                    2020-01-20</time-ago>
                            </div>
                        </h1>
                    </div>
                </div>




            </div>
            <div class="container-lg clearfix new-discussion-timeline experiment-repo-nav  p-responsive">
                <div class="repository-content ">
                    <div class="Box mt-3 position-relative">
                        <div class="Box-header py-2 d-flex flex-column flex-shrink-0 flex-md-row flex-md-items-center">
                            <div class="text-mono f6 flex-auto pr-3 flex-order-2 flex-md-order-1 mt-2 mt-md-0">
                                1682 Words
                                
                            </div>
                        </div>

                        <div id="readme" class="Box-body readme blob instapaper_body js-code-block-container">
                            <article class="markdown-body entry-content p-3 p-md-6" itemprop="text"><blockquote>
<p>原始测序文件fq.gz经genome比对，变异检测得到vcf文件，亦即变异位点文件， 只是通过看vcf文件我们是不知道些变异位点到底是位于基因的exon、intron、UTR等的哪些区域的 ，所以需要注释， 常用的vcf注释软件有annovar和snpEff 。</p>
</blockquote>
<h3 id="annovar">ANNOVAR</h3>
<p><a href="http://annovar.openbioinformatics.org/en/latest/">annovar官网</a></p>
<h4 id="安装">安装</h4>
<p>官网注册下载即可，程序为perl脚本，</p>
<h4 id="annovar结构">ANNOVAR结构</h4>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">ANNOVAR  
│  annotate_variation.pl <span style="color:#75715e">#主程序，功能包括下载数据库，三种不同的注释</span>
│  coding_change.pl <span style="color:#75715e">#可用来推断蛋白质序列</span>
│  convert2annovar.pl <span style="color:#75715e">#将多种格式转为.avinput的程序</span>
│  retrieve_seq_from_fasta.pl <span style="color:#75715e">#用于自行建立其他物种的转录本</span>
│  table_annovar.pl <span style="color:#75715e">#注释程序，可一次性完成三种类型的注释</span>
│  variants_reduction.pl <span style="color:#75715e">#可用来更灵活地定制过滤注释流程</span>
│
├─example <span style="color:#75715e">#存放示例文件</span>
│
└─humandb <span style="color:#75715e">#人类注释数据库</span>
</code></pre></div><h4 id="annovar下载数据库">ANNOVAR下载数据库</h4>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">Perl annotate_variation.pl -buildver hg19 -downdb -webfrom annovar refGene humandb/
<span style="color:#75715e">#可用-downdb avdblist参数查看可供下载数据库列表</span>
<span style="color:#75715e"># -buildver 表示version</span>
<span style="color:#75715e"># -downdb 下载数据库的指令</span>
<span style="color:#75715e"># -webfrom annovar 从annovar提供的镜像下载，不加此参数将寻找数据库本身的源</span>
<span style="color:#75715e"># humandb/ 存放于humandb/目录下</span>
<span style="color:#75715e"># 默认使用gene-based注释类型以及refGene数据库，即基于基因的注释</span>
</code></pre></div><p>基于区域的注释(Region-based Annotation):揭示variant与不同基因组特定段的关系，例如：它是否落在已知的保守基因组区域。基于区域的注释的数据库一般由UCSC提供;在一个数据库中，它不在乎位置的精确匹配，它不在乎核苷酸的识别。</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">annotate_variation.pl -regionanno -build hg19 -out ex1 -dbtype phastConsElements46way example/ex1.avinput humandb/
<span style="color:#75715e"># -regionanno 表示使用基于区域的注释</span>
<span style="color:#75715e"># -dbtype phastConsElements46way 表示使用&#34;phastConsElements46way&#34;数据库，注意需要使用Region-based的数据库</span>
</code></pre></div><p>Filter-based Annotation（基于过滤的注释）:filter-based和region-based主要的区别是，filter-based针对mutation（核苷酸的变化）而region-based针对染色体上的位置。例如region-based比对chr1:1000-1000而filter-based比对chr1:1000-1000上的A-&gt;G\</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">nnotate_variation.pl -filter -dbtype 1000g2012apr_eur -buildver hg19 -out ex1 example/ex1.avinput humandb/
<span style="color:#75715e"># -filter 使用基于过滤的注释</span>
<span style="color:#75715e"># -dbtype 1000g2012apr_eur 使用&#34;1000g2012apr_eur&#34;数据库</span>
<span style="color:#75715e">#运行命令后，已知的变异会被写入一个*dropped结尾的文件，而没有在数据库中找到的变异将会被写入*filtered结尾的文件，*dropped文件是我们所需要的结果</span>
</code></pre></div><h4 id="vcf转avinput">vcf转.avinput</h4>
<p>ANNOVAR使用.avinput格式，最重要为前5列</p>
<ul>
<li>染色体(Chromosome)</li>
<li>起始位置(Start)</li>
<li>结束位置(End)</li>
<li>参考等位基因(Reference Allele)</li>
<li>替代等位基因(Alternative Allele)</li>
<li>剩下为注释部分（可选）</li>
</ul>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">convert2annovar.pl -format vcf4 NL190929.vcf &gt;out.avinput
<span style="color:#75715e">#根据前五列去重</span>
awk <span style="color:#e6db74">&#39;!a[$1,$2,$3,$4,$5]++{print $0}&#39;</span> out.avinput &gt;uniq.avinput
</code></pre></div><h4 id="annovar注释功能">ANNOVAR注释功能</h4>
<p>用table_annovar.pl进行注释</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash">table_annovar.pl uniq.avinput ~/software/annovar/humandb -buildver hg19 -out annoresult -remove -protocol refGene,avsnp150,1000g2015aug_all,1000g2015aug_eas,1000g2015aug_sas,1000g2015aug_amr,1000g2015aug_afr,1000g2015aug_eur,gnomad_exome_20190125,dbnsfp30a,intervar_20170202,clinvar_20190305 -operation g,f,f,f,f,f,f,f,f,f,f,f 2&gt;annovar.log &amp;wait
<span style="color:#75715e"># -buildver hg19 表示使用hg19版本</span>
<span style="color:#75715e"># -out annoresult 表示输出文件的前缀为annoresult</span>
<span style="color:#75715e"># -remove 表示删除注释过程中的临时文件，试了下不加，多产生27个文件，其中annoresult.hg19_multianno.txt文件为后续分析使用</span>
<span style="color:#75715e"># -protocol 表示注释使用的数据库，用逗号隔开，且要注意顺序</span>
<span style="color:#75715e"># -operation 表示对应顺序的数据库的类型（g代表gene-based、r代表region-based、f代表filter-based），用逗号隔开，注意顺序</span>
<span style="color:#75715e"># -nastring . 表示用点号替代缺省的值</span>
<span style="color:#75715e"># -csvout 表示最后输出.csv文件</span>
<span style="color:#75715e">#2&gt;annovar.log 将标准错误输出到文件中</span>
<span style="color:#75715e">#wait等待所有线程的程序运行完执行下一步操作</span>
</code></pre></div><h4 id="intervar">InterVar</h4>
<p><a href="https://github.com/WGLab/InterVar">InterVar</a>
这个软件是同个作者写的，是增加Intervar注释的软件。</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-bash" data-lang="bash"><span style="color:#75715e">#按照参考的配置文件配好相关路径</span>
cp ~/config.ini ./ 
python2 ~/InterVar-master/Intervar.py -c ./config.ini 2&gt;Intervar.log
<span style="color:#75715e">#从其运行来看，是对数据库Intervar再次运行脚本annotate_variation.pl生成annocar数据库，convert2annovar.pl ,table_annovar.pl产生三个文件，其中</span>
intervar_result.hg19_multianno.txt.intervar为后续分析使用，
<span style="color:#75715e">##去重</span>
awk <span style="color:#e6db74">&#39;!a[$1,$2,$3,$4,$5]++{print $0}&#39;</span> intervar_result.hg19_multianno.txt.intervar &gt;intervar_result_uniq
</code></pre></div></article>
                        </div>
                    </div>
                </div>
            </div>
        </main>
    </div>
</div>

<div class="utterances"></div>

<script>
    var systemThemeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
    var systemTheme = systemThemeMediaQuery.matches ? 'dark' : 'light';
    var utterancesTheme = systemTheme === 'dark' ? 'github-dark' : 'github-light';
    var utterancesScript = document.createElement('script');
    utterancesScript.src = 'https://utteranc.es/client.js';
    utterancesScript.setAttribute('repo', 'liripo\/liripo.github.io');
    utterancesScript.setAttribute('issue-term', 'title');
    utterancesScript.setAttribute('theme', utterancesTheme);
    utterancesScript.setAttribute('crossorigin', 'anonymous');
    utterancesScript.setAttribute('async', '');
    document.querySelector('.utterances').appendChild(utterancesScript);
</script>
<noscript>Please enable JavaScript to view the comments</noscript>


  <div class="footer container-lg width-full p-responsive" role="contentinfo">
    <div
        class="position-relative d-flex flex-row-reverse flex-lg-row flex-wrap flex-lg-nowrap flex-justify-center flex-lg-justify-between pt-6 pb-2 mt-6 f6 text-gray border-top border-gray-light ">
        <ul
            class="list-style-none d-flex flex-wrap col-12 col-lg-5 flex-justify-center flex-lg-justify-between mb-2 mb-lg-0">
            <li class="mr-3 mr-lg-0">Liripo</a></li>
        </ul>

        <a aria-label="Homepage" title="Liripo" class="footer-octicon d-none d-lg-block mx-lg-4"
            href="https://liripo.github.io">
            <svg height="24" class="octicon octicon-mark-github" viewBox="0 0 16 16" version="1.1" width="24"
                aria-hidden="true">
                <path fill-rule="evenodd"
                    d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z">
                </path>
            </svg>
        </a>
        <ul
            class="list-style-none d-flex flex-wrap col-12 col-lg-5 flex-justify-center flex-lg-justify-between mb-2 mb-lg-0">
        </ul>
    </div>
    <div class="d-flex flex-justify-center pb-6">
        <span class="f6 text-gray-light"></span>
    </div>
</div>

<script crossorigin="anonymous"
    type="application/javascript" src="https://liripo.github.io/js/frameworks.js"></script>

<script crossorigin="anonymous"
    type="application/javascript" src="https://liripo.github.io/js/github-bootstrap.js"></script>
<script src="https://liripo.github.io/js/stop.js"></script>
<script src="https://liripo.github.io/js/contributions.js"></script>

</body>

</html>