<!DOCTYPE html>
<html lang="Zh-cn">
<head>
    <meta charset="utf-8">
    <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://liripo.github.io/css/frameworks.css" />
    <link crossorigin="anonymous" media="all" rel="stylesheet" href="https://liripo.github.io/css/github.css" />
    
    <link crossorigin="anonymous" media="(prefers-color-scheme: dark)" rel="stylesheet"
        href="https://liripo.github.io/css/dark.css" />
    
    <meta name="viewport" content="width=device-width">
    <meta name="description"
        content="一般已发表文章所包含的数据可以在NCBI (SRA、GEO 等)、EMBL-EBI 等相关数据库获得。 ncbi数据下载参考 高通量测序知识 下载测序数" />
    <meta name="keywords"
        content="" />
    <meta name="robots" content="noodp" />
    <link rel="canonical" href="https://liripo.github.io/posts/2020/rna-seq%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B/" />

    
    <meta name="twitter:card" content="summary" />
    <meta name="twitter:title" content="RNA-seq分析流程 - Liripo" />
    <meta name="twitter:description"
        content="一般已发表文章所包含的数据可以在NCBI (SRA、GEO 等)、EMBL-EBI 等相关数据库获得。 ncbi数据下载参考 高通量测序知识 下载测序数" />
    <meta name="twitter:site" content="https://liripo.github.io" />
    <meta name="twitter:creator" content="" />
    <meta name="twitter:image"
        content="https://liripo.github.io">

    
    <meta property="og:type" content="article" />
    <meta property="og:title" content="RNA-seq分析流程 - Liripo">
    <meta property="og:description"
        content="一般已发表文章所包含的数据可以在NCBI (SRA、GEO 等)、EMBL-EBI 等相关数据库获得。 ncbi数据下载参考 高通量测序知识 下载测序数" />
    <meta property="og:url" content="https://liripo.github.io/posts/2020/rna-seq%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B/" />
    <meta property="og:site_name" content="RNA-seq分析流程" />
    <meta property="og:image"
        content="https://liripo.github.io">
    <meta property="og:image:width" content="2048">
    <meta property="og:image:height" content="1024">
    
    <meta property="article:published_time" content="2020-10-18 00:00:00 &#43;0000 UTC" />
    

    
    

    
    

    <title>RNA-seq分析流程 - Liripo</title>

    <link rel="icon" type="image/x-icon" class="js-site-favicon" href="https://liripo.github.io/images/favicon.ico">
    <meta name="theme-color" content="#1e2327">

    
    

    
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'UA-155993181-1', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>

</head>

<body class="env-production emoji-size-boost page-responsive page-profile">
  <div class="position-relative js-header-wrapper ">
    <span class="Progress progress-pjax-loader position-fixed width-full js-pjax-loader-bar">
        <span class="progress-pjax-loader-bar top-0 left-0" style="width: 0%;"></span>
    </span>
    <header class="Header py-lg-0 js-details-container Details flex-wrap flex-lg-nowrap p-responsive" role="banner">
        <div class="Header-item d-none d-lg-flex">
            <a class="Header-link" href="https://liripo.github.io" aria-label="Homepage"
                data-ga-click="Header, go to dashboard, icon:logo">
                <svg class="octicon octicon-mark-github v-align-middle" height="32" viewBox="0 0 16 16" version="1.1"
                    width="32" aria-hidden="true">
                    <path fill-rule="evenodd"
                        d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z" />
                </svg>
            </a>
        </div>
        <div
            class="Header-item Header-item--full flex-column flex-lg-row width-full flex-order-2 flex-lg-order-none mr-0 mr-lg-3 mt-3 mt-lg-0 Details-content--hidden">
            <div class="header-search flex-self-stretch flex-lg-self-auto mr-0 mr-lg-3 mb-3 mb-lg-0 scoped-search site-scoped-search js-site-search position-relative js-jump-to"
                role="combobox" aria-owns="jump-to-results" aria-label="Search or jump to" aria-haspopup="listbox"
                aria-expanded="false">
                <div class="position-relative">
                </div>
            </div>
        </div>
        <nav class="d-flex" aria-label="Global">
            <a class="js-selected-navigation-item Header-link py-lg-3">&nbsp;</a>
        </nav>
        <div class="Header-item Header-item--full flex-justify-center d-lg-none position-relative" style="margin-right: auto;">
            <a class="Header-link" href="https://liripo.github.io" aria-label="Homepage"
                data-ga-click="Header, go to dashboard, icon:logo">
                <svg class="octicon octicon-mark-github v-align-middle" height="32" viewBox="0 0 16 16" version="1.1"
                    width="32" aria-hidden="true">
                    <path fill-rule="evenodd"
                        d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z" />
                </svg>
            </a>
        </div>
        <div class="Header-item position-relative mr-0 d-none d-lg-flex">
            <details class="details-overlay details-reset">
                <summary class="Header-link" aria-label="View profile and more"
                    data-ga-click="Header, show menu, icon:avatar">
                    
                    <img alt="" class="avatar" src="https://themes.gohugo.io/theme/github-style/images/avatar.png" height="20" width="20">
                    
                </summary>
            </details>
        </div>
    </header>
</div>

<div id="start-of-content" class="show-on-focus"></div>
<div id="js-flash-container">
</div>

  
<div class="application-main " data-commit-hovercards-enabled="">
    <div itemscope="" itemtype="http://schema.org/SoftwareSourceCode" class="">
        <main>
            <div class="pagehead repohead instapaper_ignore readability-menu experiment-repo-nav pt-0 pt-lg-4 ">
                <div class="repohead-details-container clearfix container-lg p-responsive d-none d-lg-block">
                    <div class="mb-3 d-flex">
                        <h1 class="public css-truncate float-none flex-auto width-fit pl-0">
                            <a class="avatar mr-1" href="https://liripo.github.io/about/">
                                

                                <img src="https://themes.gohugo.io/theme/github-style/images/avatar.png" width="26" height="26">
                                
                            </a>
                            <span class="author"><a href="https://liripo.github.io">liripo</a></span>
                            <span class="path-divider">/</span>
                            <strong itemprop="name" class="css-truncate-target" style="max-width: 410px"><a
                                    href="https://liripo.github.io/posts/2020/rna-seq%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B/">RNA-seq分析流程</a></strong>

                            <div class="d-block text-small text-gray">
                                Created <time-ago datetime="2020-10-18" class="no-wrap"
                                    title="Created at 2020/10/18">
                                    2020-10-18</time-ago>
                                <span class="file-info-divider"></span>
                                Modified <time-ago datetime="2020-10-18" class="no-wrap"
                                    title="Modified  at 2020/10/18">
                                    2020-10-18</time-ago>
                            </div>
                        </h1>
                    </div>
                </div>




            </div>
            <div class="container-lg clearfix new-discussion-timeline experiment-repo-nav  p-responsive">
                <div class="repository-content ">
                    <div class="Box mt-3 position-relative">
                        <div class="Box-header py-2 d-flex flex-column flex-shrink-0 flex-md-row flex-md-items-center">
                            <div class="text-mono f6 flex-auto pr-3 flex-order-2 flex-md-order-1 mt-2 mt-md-0">
                                2204 Words
                                
                            </div>
                        </div>

                        <div id="readme" class="Box-body readme blob instapaper_body js-code-block-container">
                            <article class="markdown-body entry-content p-3 p-md-6" itemprop="text"><blockquote>
<p>一般已发表文章所包含的数据可以在NCBI (SRA、GEO 等)、EMBL-EBI 等相关数据库获得。</p>
</blockquote>
<p><a href="https://liripo.github.io/post/ncbi%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD/">ncbi数据下载参考</a></p>
<p><a href="https://shixiangwang.github.io/home/cn/post/2017-10-27-rnaseq-data-analysis/">高通量测序知识</a></p>
<p>下载测序数据后，可以进行质控(fastqc等)，比对(bwa,STAT,subread等，非常多)，获取counts数，之后差异分析，GO，KEGG等。当然还有call snp等。</p>
<h1 id="质控">质控</h1>
<p><a href="https://liripo.github.io/post/fastqc%E4%BD%BF%E7%94%A8/">fastqc使用</a>，相对应的R包<a href="http://www.sthda.com/english/wiki/fastqcr-an-r-package-facilitating-quality-controls-of-sequencing-data-for-large-numbers-of-samples">fastqcr</a>，</p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/Rqc.html">rqc</a></p>
<p><a href="https://liripo.github.io/post/fastp-gencore%E4%BD%BF%E7%94%A8/">fastp</a></p>
<p>Biostrings包计算GC含量，Q20等</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-R" data-lang="R"><span style="color:#a6e22e">library</span>(Biostrings)
filepath <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">system.file</span>(<span style="color:#e6db74">&#34;extdata&#34;</span>, <span style="color:#e6db74">&#34;s_1_sequence.txt&#34;</span>,
                        package<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;Biostrings&#34;</span>)
qdna2 <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">readQualityScaledDNAStringSet</span>(filepath)
qdna2
<span style="color:#75715e">#得到每一行的GC含量</span>
GC_content <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">letterFrequency</span>(<span style="color:#a6e22e">DNAStringSet</span>(qdna2), <span style="color:#66d9ef">letters</span><span style="color:#f92672">=</span><span style="color:#e6db74">&#34;CG&#34;</span>,as.prob <span style="color:#f92672">=</span> T)
<span style="color:#75715e">#得到整个fastq的GC含量</span>
<span style="color:#a6e22e">letterFrequency</span>(<span style="color:#a6e22e">DNAStringSet</span>(qdna2), <span style="color:#66d9ef">letters</span><span style="color:#f92672">=</span><span style="color:#e6db74">&#34;CG&#34;</span>,as.prob <span style="color:#f92672">=</span> T,collapse <span style="color:#f92672">=</span> T)
<span style="color:#75715e">#Q20，将质量分数转为数值去计算,整个文件</span>
qa <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">as</span>(<span style="color:#a6e22e">quality</span>(qdna2),<span style="color:#e6db74">&#34;IntegerList&#34;</span>)
<span style="color:#a6e22e">library</span>(dplyr)
sum_20 <span style="color:#f92672">&lt;-</span> purrr<span style="color:#f92672">::</span><span style="color:#a6e22e">map_int</span>(<span style="color:#a6e22e">seq_along</span>(qa),<span style="color:#f92672">~</span><span style="color:#a6e22e">length</span>(<span style="color:#a6e22e">which</span>(qa[[.x]]<span style="color:#f92672">&gt;</span><span style="color:#ae81ff">20</span>)) <span style="color:#f92672">%&gt;%</span> 
	<span style="color:#a6e22e">sum</span>()
sum_all <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">quality</span>(qdna2) <span style="color:#f92672">%&gt;%</span> <span style="color:#a6e22e">width</span>() <span style="color:#f92672">%&gt;%</span> <span style="color:#a6e22e">sum</span>()
q20 <span style="color:#f92672">&lt;-</span> sum_20<span style="color:#f92672">/</span>sum_all
</code></pre></div><h1 id="比对">比对</h1>
<p>在比对文件之前需要先要去掉adapter，得到不含有adapter 的fastq 文件。有一个flexbark可以，</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">FLEXBAR -t 输出文件名 -qf i1.8 -r 输入文件 -a Adapters.fa
</code></pre></div><p>我没试过去接头。</p>
<blockquote>
<p>反正有了trimmed 数据后，就要与参考文件进行比对。目前大家所认同的比对算法
有很多，比如Bowtie2，BWA，TopHat，HISAT2 和STAR 等等。试下
综合指标 (Sahraeian 等，2017) 比较高的STAR 进行比对。</p>
</blockquote>
<p>比对大部分需要建立索引,加快速度。代码</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">STAR --runThreadN 线程数 --runMode genomeGenerate
--genomeDir 存放参考基因的目录 --genomeFastaFiles 参考基因组的
fasta 文件 --sjdbGTFfile 参考基因组的gtf 文件 --sjdbOverhang
索引长度,这个是reads长度的最大值减1，默认是100
</code></pre></div><p>建立好索引就可以对数据进行比对了，比对如下：</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">STAR --genomeDir 存放参考基因的目录 --readFilesIn 输入文件<span style="color:#f92672">(</span>read1,if need;read2<span style="color:#f92672">)</span>
--runThreadN 线程数 --outSAMtype BAM --outFileNamePrefix 输
出文件前缀
</code></pre></div><p>具体可以参考<a href="https://www.bioinfo-scrounger.com/archives/288/">https://www.bioinfo-scrounger.com/archives/288/</a></p>
<h1 id="counts数">counts数</h1>
<ul>
<li>bedtools multicov</li>
</ul>
<p><a href="https://liripo.github.io/post/%E7%94%9F%E4%BF%A1%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F%E6%B1%87%E6%80%BB/#gtf/gff3/GenePred">bed文件从gtf的获取</a></p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">bedtools multicov -bams *.bam -bed bed文件 &gt;reads_by_bed.txt
</code></pre></div><p>虽然-bams可以使用通配符，但是我一般还是用shell写脚本循环，因为输出文件没有列名，很容易混淆的。我认为输出文件最好能带个列名。</p>
<ul>
<li><strong><a href="http://bioinformatics.oxfordjournals.org/cgi/reprint/btt656?ijkey=ZzPz96t2lqzAH6F&amp;keytype=ref">featureCounts</a></strong></li>
</ul>
<p><a href="http://bioinf.wehi.edu.au/featureCounts/">user guide</a></p>
<p>比如双端测序，</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">featureCounts -p -t gene -g gene_name -a annotation.gtf -o counts.txt *_mapping__PE.bam
</code></pre></div><p>直接使用gene_name做标识符也行,提取出gene的位置。</p>
<p>类似的还有Htseq-count</p>
<p>而samtools 的bedcov计算的则不同，是bed文件每个区域的深度和。即samtools depth的和。</p>
<p>而计算覆盖到整个参考fasta可以用samtools flagstat或者更详细的samtools idxstats。虽然featureCounts等也会给出，不过是输出在标准输出里的，不在输出文件里。</p>
<h2 id="tpm等转化">TPM等转化</h2>
<p>reads数即counts数，接着如果需要比较不同样本同个基因上的表达丰度情况，则需要对count数进行标准化。</p>
<blockquote>
<p>落在一个基因区域内的read counts数目一般可以认为取决于length of the gene（基因长度）和sequencing depth（测序深度）</p>
</blockquote>
<ul>
<li>RPKM和FPKM。前者是以每个reads作为一个单位，在单端测序中应用较多；而后者是以fragment作为一个单位，主要应用在双端测序后的分析。</li>
<li>最近几年开始应用TPM（Transcripts Per Million）较多，Transcripts Per Kilobase of exonmodel per Million mapped reads (即是先对reads进行基因长度矫正，然后再除以所有矫正后的reads的总和)，认为优化的RPKM计算方法，可以用于同一物种不同组织的比较。</li>
</ul>
<blockquote>
<p>FPKM：Fragments Per Kilobase of transcript per Million fragments mapped，每1百万个map上的reads中map到map到外显子的每1k个碱基上的fragments个数。如果一对paired reads都比对上了，那么这对reads当做一个fragments；如果paired reads中一个比对上，另外一个没比对上，那么比对上的那个reads当做一个fragments。</p>
</blockquote>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">FPKM<span style="color:#f92672">=</span>read counts/<span style="color:#f92672">(</span>mapped reads <span style="color:#f92672">(</span>Millions<span style="color:#f92672">)</span> * exon length<span style="color:#f92672">(</span>KB<span style="color:#f92672">))</span>
</code></pre></div><p>所以，这个fragments是不是featureCounts算出的counts数我不太确定。反正rpkm肯定是。</p>
<p>在R中，则</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-R" data-lang="R">counts_to_fpkm <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(counts,lengths){
    rates <span style="color:#f92672">&lt;-</span> counts<span style="color:#f92672">/</span>lengths
    fpkm <span style="color:#f92672">&lt;-</span> rates <span style="color:#f92672">/</span><span style="color:#a6e22e">sum</span>(counts) <span style="color:#f92672">*</span> <span style="color:#ae81ff">1e9</span>
}
</code></pre></div><p>而TPM则是</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-R" data-lang="R">counts_to_tpm <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(counts,lengths){
    rates <span style="color:#f92672">&lt;-</span> counts<span style="color:#f92672">/</span>lengths
    tpm <span style="color:#f92672">&lt;-</span> rates<span style="color:#f92672">/</span><span style="color:#a6e22e">sum</span>(rates) <span style="color:#f92672">*</span> <span style="color:#ae81ff">10</span>^6
}
</code></pre></div><p>有人认为是clean reads，也就是比对前的clean reads总数【我觉得没必要】，但大多数人还是定义为有效的reads，即mapped reads。那么，mapped reads究竟是哪个值呢，因为比对到gene,exon的总mapped reads不等于map到fasta的reads吧，毕竟肯定有内含子之类的。这里我觉得既然是标准化，就应该是比如feauture得到的那些数据去标准就好，所以比值应该是比对到gene,exon的总mapped reads【我认为】。</p>
<p>那么，exon length呢，粗暴地理解为gene length？【这样方便获取】，正常来说应该是每个gene id的exon的长度总和。获取所有外显子的长度和呢，一般可以从gtf文件入手，但是gtf文件中的exon的区域很多都是重合的（不同转录本的可变剪切）【比较麻烦。】</p>
<p>当然，使用featureCounts时，可以</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">featureCounts -p -t exon -g gene_id -a annotation.gtf -o counts.txt *_mapping__PE.bam
</code></pre></div><p>输出文件有一列length，即为exon的length,方便。</p>
<p>而在R中,可以使用GenomicFeatures计算每个基因下所有外显子的总长度</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-R" data-lang="R"><span style="color:#75715e">#制作txdb对象</span>
<span style="color:#a6e22e">library</span>(GenomicFeatures)
txdb <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">makeTxDbFromGFF</span>(<span style="color:#e6db74">&#34;hg38.gtf&#34;</span>,format<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;gtf&#34;</span>)
<span style="color:#75715e">#通过exonsBy获取每个gene上的所有外显子的起始位点和终止位点，然后用reduce去除掉重叠冗余的部分，最后计算长度</span>
exons_gene <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">exonsBy</span>(txdb, by <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;gene&#34;</span>)
exons_gene_lens <span style="color:#f92672">&lt;-</span> purrr<span style="color:#f92672">::</span><span style="color:#a6e22e">map_int</span>(exons_gene,<span style="color:#f92672">~</span><span style="color:#a6e22e">sum</span>(<span style="color:#a6e22e">width</span>(<span style="color:#a6e22e">reduce</span>(.x))))
</code></pre></div><h1 id="差异分析">差异分析</h1>
<p>差异分析使用的是counts文件，相关性分析之类的可以使用TPM数据。</p>
<h1 id="call-snp">call snp</h1>
</article>
                        </div>
                    </div>
                </div>
            </div>
        </main>
    </div>
</div>

<div class="utterances"></div>

<script>
    var systemThemeMediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
    var systemTheme = systemThemeMediaQuery.matches ? 'dark' : 'light';
    var utterancesTheme = systemTheme === 'dark' ? 'github-dark' : 'github-light';
    var utterancesScript = document.createElement('script');
    utterancesScript.src = 'https://utteranc.es/client.js';
    utterancesScript.setAttribute('repo', 'liripo\/liripo.github.io');
    utterancesScript.setAttribute('issue-term', 'title');
    utterancesScript.setAttribute('theme', utterancesTheme);
    utterancesScript.setAttribute('crossorigin', 'anonymous');
    utterancesScript.setAttribute('async', '');
    document.querySelector('.utterances').appendChild(utterancesScript);
</script>
<noscript>Please enable JavaScript to view the comments</noscript>


  <div class="footer container-lg width-full p-responsive" role="contentinfo">
    <div
        class="position-relative d-flex flex-row-reverse flex-lg-row flex-wrap flex-lg-nowrap flex-justify-center flex-lg-justify-between pt-6 pb-2 mt-6 f6 text-gray border-top border-gray-light ">
        <ul
            class="list-style-none d-flex flex-wrap col-12 col-lg-5 flex-justify-center flex-lg-justify-between mb-2 mb-lg-0">
            <li class="mr-3 mr-lg-0">Liripo</a></li>
        </ul>

        <a aria-label="Homepage" title="Liripo" class="footer-octicon d-none d-lg-block mx-lg-4"
            href="https://liripo.github.io">
            <svg height="24" class="octicon octicon-mark-github" viewBox="0 0 16 16" version="1.1" width="24"
                aria-hidden="true">
                <path fill-rule="evenodd"
                    d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z">
                </path>
            </svg>
        </a>
        <ul
            class="list-style-none d-flex flex-wrap col-12 col-lg-5 flex-justify-center flex-lg-justify-between mb-2 mb-lg-0">
        </ul>
    </div>
    <div class="d-flex flex-justify-center pb-6">
        <span class="f6 text-gray-light"></span>
    </div>
</div>

<script crossorigin="anonymous"
    type="application/javascript" src="https://liripo.github.io/js/frameworks.js"></script>

<script crossorigin="anonymous"
    type="application/javascript" src="https://liripo.github.io/js/github-bootstrap.js"></script>
<script src="https://liripo.github.io/js/stop.js"></script>
<script src="https://liripo.github.io/js/contributions.js"></script>

</body>

</html>