<!DOCTYPE html>
<html lang="Zh-cn">

  <head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>RNA-seq分析流程 | Liripo</title>
    <link rel="shortcut icon" href="../../image/title.svg" type="image/x-icon">
  <script src="../../fontawesome-free-5.11.2-web/fontawesome-free-5.11.2-web/js/all.js" crossorigin="anonymous"></script>

  <style>
  	:root {
  		--pagecolour: #333333;
  		--maintext: #dddddd;
  		--faded: #888888;
  		--highlight: #6BB0DF;
  		--lowlight: #ba68c8;
	}
  </style>
  <link rel="stylesheet" href="../../css/et_book.css" />
  <link rel="stylesheet" href="../../css/search.css" />
  <link rel="stylesheet" href="../../css/style.css" />
  <link rel="alternate" type="application/rss+xml" href="https://liripo.github.io/index.xml" title="Liripo">
  
  <link rel="stylesheet" href="https://fonts.loli.net/css?family=Tangerine">

  
<meta name="twitter:card" content="summary">
<meta property="og:title" content="RNA-seq分析流程">




</head>


  <body>
    



    <div class="main-content">
      <nav>
  <ul class="menu">
    
    <li><a href="../../">Home</a></li>
    
    <li><a href="../../about/">About</a></li>
    
    <li><a href="../../post/">Posts</a></li>
    
    <li><a href="../../tags/">Tags</a></li>
    
  </ul>
</nav>

      <div>
  <h1>RNA-seq分析流程</h1>
  <h3>
    18 Oct 2020
  </h3>
   <h5 id="wc">2204 Words|Read in about 5       Min|本文总阅读量<span id="busuanzi_value_page_pv"></span>次</h5>
    </h5>
  <br><br>
</div>


      <main>
        

<aside>
  <div id='anchors-navbar'>
  <i class='fas fa-anchor'></i>
  <nav id="TableOfContents">
  <ul>
    <li><a href="#heading">质控</a></li>
    <li><a href="#heading-1">比对</a></li>
    <li><a href="#counts">counts数</a>
      <ul>
        <li><a href="#tpm">TPM等转化</a></li>
      </ul>
    </li>
    <li><a href="#heading-2">差异分析</a></li>
    <li><a href="#call-snp">call snp</a></li>
  </ul>
</nav>
  </div>
</aside>

        <blockquote>
<p>一般已发表文章所包含的数据可以在NCBI (SRA、GEO 等)、EMBL-EBI 等相关数据库获得。</p>
</blockquote>
<p><a href="https://liripo.github.io/post/ncbi%E6%95%B0%E6%8D%AE%E4%B8%8B%E8%BD%BD/">ncbi数据下载参考</a></p>
<p><a href="https://shixiangwang.github.io/home/cn/post/2017-10-27-rnaseq-data-analysis/">高通量测序知识</a></p>
<p>下载测序数据后，可以进行质控(fastqc等)，比对(bwa,STAT,subread等，非常多)，获取counts数，之后差异分析，GO，KEGG等。当然还有call snp等。</p>
<h1 id="heading">质控</h1>
<p><a href="https://liripo.github.io/post/fastqc%E4%BD%BF%E7%94%A8/">fastqc使用</a>，相对应的R包<a href="http://www.sthda.com/english/wiki/fastqcr-an-r-package-facilitating-quality-controls-of-sequencing-data-for-large-numbers-of-samples">fastqcr</a>，</p>
<p><a href="https://bioconductor.org/packages/release/bioc/html/Rqc.html">rqc</a></p>
<p><a href="https://liripo.github.io/post/fastp-gencore%E4%BD%BF%E7%94%A8/">fastp</a></p>
<p>Biostrings包计算GC含量，Q20等</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-R" data-lang="R"><span style="color:#a6e22e">library</span>(Biostrings)
filepath <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">system.file</span>(<span style="color:#e6db74">&#34;</span><span style="color:#e6db74">extdata&#34;</span>, <span style="color:#e6db74">&#34;</span><span style="color:#e6db74">s_1_sequence.txt&#34;</span>,
                        package<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;</span><span style="color:#e6db74">Biostrings&#34;</span>)
qdna2 <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">readQualityScaledDNAStringSet</span>(filepath)
qdna2
<span style="color:#75715e">#得到每一行的GC含量</span>
GC_content <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">letterFrequency</span>(<span style="color:#a6e22e">DNAStringSet</span>(qdna2), <span style="color:#66d9ef">letters</span><span style="color:#f92672">=</span><span style="color:#e6db74">&#34;</span><span style="color:#e6db74">CG&#34;</span>,as.prob <span style="color:#f92672">=</span> T)
<span style="color:#75715e">#得到整个fastq的GC含量</span>
<span style="color:#a6e22e">letterFrequency</span>(<span style="color:#a6e22e">DNAStringSet</span>(qdna2), <span style="color:#66d9ef">letters</span><span style="color:#f92672">=</span><span style="color:#e6db74">&#34;</span><span style="color:#e6db74">CG&#34;</span>,as.prob <span style="color:#f92672">=</span> T,collapse <span style="color:#f92672">=</span> T)
<span style="color:#75715e">#Q20，将质量分数转为数值去计算,整个文件</span>
qa <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">as</span>(<span style="color:#a6e22e">quality</span>(qdna2),<span style="color:#e6db74">&#34;</span><span style="color:#e6db74">IntegerList&#34;</span>)
<span style="color:#a6e22e">library</span>(dplyr)
sum_20 <span style="color:#f92672">&lt;-</span> purrr<span style="color:#f92672">::</span><span style="color:#a6e22e">map_int</span>(<span style="color:#a6e22e">seq_along</span>(qa),<span style="color:#f92672">~</span><span style="color:#a6e22e">length</span>(<span style="color:#a6e22e">which</span>(qa[[.x]]<span style="color:#f92672">&gt;</span><span style="color:#ae81ff">20</span>)) <span style="color:#f92672">%&gt;%</span> 
	<span style="color:#a6e22e">sum</span>()
sum_all <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">quality</span>(qdna2) <span style="color:#f92672">%&gt;%</span> <span style="color:#a6e22e">width</span>() <span style="color:#f92672">%&gt;%</span> <span style="color:#a6e22e">sum</span>()
q20 <span style="color:#f92672">&lt;-</span> sum_20<span style="color:#f92672">/</span>sum_all
</code></pre></div><h1 id="heading-1">比对</h1>
<p>在比对文件之前需要先要去掉adapter，得到不含有adapter 的fastq 文件。有一个flexbark可以，</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">FLEXBAR -t 输出文件名 -qf i1.8 -r 输入文件 -a Adapters.fa
</code></pre></div><p>我没试过去接头。</p>
<blockquote>
<p>反正有了trimmed 数据后，就要与参考文件进行比对。目前大家所认同的比对算法
有很多，比如Bowtie2，BWA，TopHat，HISAT2 和STAR 等等。试下
综合指标 (Sahraeian 等，2017) 比较高的STAR 进行比对。</p>
</blockquote>
<p>比对大部分需要建立索引,加快速度。代码</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">STAR --runThreadN 线程数 --runMode genomeGenerate
--genomeDir 存放参考基因的目录 --genomeFastaFiles 参考基因组的
fasta 文件 --sjdbGTFfile 参考基因组的gtf 文件 --sjdbOverhang
索引长度,这个是reads长度的最大值减1，默认是100
</code></pre></div><p>建立好索引就可以对数据进行比对了，比对如下：</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">STAR --genomeDir 存放参考基因的目录 --readFilesIn 输入文件<span style="color:#f92672">(</span>read1,if need;read2<span style="color:#f92672">)</span>
--runThreadN 线程数 --outSAMtype BAM --outFileNamePrefix 输
出文件前缀
</code></pre></div><p>具体可以参考<a href="https://www.bioinfo-scrounger.com/archives/288/">https://www.bioinfo-scrounger.com/archives/288/</a></p>
<h1 id="counts">counts数</h1>
<ul>
<li>bedtools multicov</li>
</ul>
<p><a href="https://liripo.github.io/post/%E7%94%9F%E4%BF%A1%E6%96%87%E4%BB%B6%E6%A0%BC%E5%BC%8F%E6%B1%87%E6%80%BB/#gtf/gff3/GenePred">bed文件从gtf的获取</a></p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">bedtools multicov -bams *.bam -bed bed文件 &gt;reads_by_bed.txt
</code></pre></div><p>虽然-bams可以使用通配符，但是我一般还是用shell写脚本循环，因为输出文件没有列名，很容易混淆的。我认为输出文件最好能带个列名。</p>
<ul>
<li><strong><a href="http://bioinformatics.oxfordjournals.org/cgi/reprint/btt656?ijkey=ZzPz96t2lqzAH6F&amp;keytype=ref">featureCounts</a></strong></li>
</ul>
<p><a href="http://bioinf.wehi.edu.au/featureCounts/">user guide</a></p>
<p>比如双端测序，</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">featureCounts -p -t gene -g gene_name -a annotation.gtf -o counts.txt *_mapping__PE.bam
</code></pre></div><p>直接使用gene_name做标识符也行,提取出gene的位置。</p>
<p>类似的还有Htseq-count</p>
<p>而samtools 的bedcov计算的则不同，是bed文件每个区域的深度和。即samtools depth的和。</p>
<p>而计算覆盖到整个参考fasta可以用samtools flagstat或者更详细的samtools idxstats。虽然featureCounts等也会给出，不过是输出在标准输出里的，不在输出文件里。</p>
<h2 id="tpm">TPM等转化</h2>
<p>reads数即counts数，接着如果需要比较不同样本同个基因上的表达丰度情况，则需要对count数进行标准化。</p>
<blockquote>
<p>落在一个基因区域内的read counts数目一般可以认为取决于length of the gene（基因长度）和sequencing depth（测序深度）</p>
</blockquote>
<ul>
<li>RPKM和FPKM。前者是以每个reads作为一个单位，在单端测序中应用较多；而后者是以fragment作为一个单位，主要应用在双端测序后的分析。</li>
<li>最近几年开始应用TPM（Transcripts Per Million）较多，Transcripts Per Kilobase of exonmodel per Million mapped reads (即是先对reads进行基因长度矫正，然后再除以所有矫正后的reads的总和)，认为优化的RPKM计算方法，可以用于同一物种不同组织的比较。</li>
</ul>
<blockquote>
<p>FPKM：Fragments Per Kilobase of transcript per Million fragments mapped，每1百万个map上的reads中map到map到外显子的每1k个碱基上的fragments个数。如果一对paired reads都比对上了，那么这对reads当做一个fragments；如果paired reads中一个比对上，另外一个没比对上，那么比对上的那个reads当做一个fragments。</p>
</blockquote>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">FPKM<span style="color:#f92672">=</span>read counts/<span style="color:#f92672">(</span>mapped reads <span style="color:#f92672">(</span>Millions<span style="color:#f92672">)</span> * exon length<span style="color:#f92672">(</span>KB<span style="color:#f92672">)</span><span style="color:#f92672">)</span>
</code></pre></div><p>所以，这个fragments是不是featureCounts算出的counts数我不太确定。反正rpkm肯定是。</p>
<p>在R中，则</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-R" data-lang="R">counts_to_fpkm <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(counts,lengths){
    rates <span style="color:#f92672">&lt;-</span> counts<span style="color:#f92672">/</span>lengths
    fpkm <span style="color:#f92672">&lt;-</span> rates <span style="color:#f92672">/</span><span style="color:#a6e22e">sum</span>(counts) <span style="color:#f92672">*</span> <span style="color:#ae81ff">1e9</span>
}
</code></pre></div><p>而TPM则是</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-R" data-lang="R">counts_to_tpm <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">function</span>(counts,lengths){
    rates <span style="color:#f92672">&lt;-</span> counts<span style="color:#f92672">/</span>lengths
    tpm <span style="color:#f92672">&lt;-</span> rates<span style="color:#f92672">/</span><span style="color:#a6e22e">sum</span>(rates) <span style="color:#f92672">*</span> <span style="color:#ae81ff">10</span>^6
}
</code></pre></div><p>有人认为是clean reads，也就是比对前的clean reads总数【我觉得没必要】，但大多数人还是定义为有效的reads，即mapped reads。那么，mapped reads究竟是哪个值呢，因为比对到gene,exon的总mapped reads不等于map到fasta的reads吧，毕竟肯定有内含子之类的。这里我觉得既然是标准化，就应该是比如feauture得到的那些数据去标准就好，所以比值应该是比对到gene,exon的总mapped reads【我认为】。</p>
<p>那么，exon length呢，粗暴地理解为gene length？【这样方便获取】，正常来说应该是每个gene id的exon的长度总和。获取所有外显子的长度和呢，一般可以从gtf文件入手，但是gtf文件中的exon的区域很多都是重合的（不同转录本的可变剪切）【比较麻烦。】</p>
<p>当然，使用featureCounts时，可以</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-shell" data-lang="shell">featureCounts -p -t exon -g gene_id -a annotation.gtf -o counts.txt *_mapping__PE.bam
</code></pre></div><p>输出文件有一列length，即为exon的length,方便。</p>
<p>而在R中,可以使用GenomicFeatures计算每个基因下所有外显子的总长度</p>
<div class="highlight"><pre style="color:#f8f8f2;background-color:#272822;-moz-tab-size:4;-o-tab-size:4;tab-size:4"><code class="language-R" data-lang="R"><span style="color:#75715e">#制作txdb对象</span>
<span style="color:#a6e22e">library</span>(GenomicFeatures)
txdb <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">makeTxDbFromGFF</span>(<span style="color:#e6db74">&#34;</span><span style="color:#e6db74">hg38.gtf&#34;</span>,format<span style="color:#f92672">=</span><span style="color:#e6db74">&#34;</span><span style="color:#e6db74">gtf&#34;</span>)
<span style="color:#75715e">#通过exonsBy获取每个gene上的所有外显子的起始位点和终止位点，然后用reduce去除掉重叠冗余的部分，最后计算长度</span>
exons_gene <span style="color:#f92672">&lt;-</span> <span style="color:#a6e22e">exonsBy</span>(txdb, by <span style="color:#f92672">=</span> <span style="color:#e6db74">&#34;</span><span style="color:#e6db74">gene&#34;</span>)
exons_gene_lens <span style="color:#f92672">&lt;-</span> purrr<span style="color:#f92672">::</span><span style="color:#a6e22e">map_int</span>(exons_gene,<span style="color:#f92672">~</span><span style="color:#a6e22e">sum</span>(<span style="color:#a6e22e">width</span>(<span style="color:#a6e22e">reduce</span>(.x))))
</code></pre></div><h1 id="heading-2">差异分析</h1>
<p>差异分析使用的是counts文件，相关性分析之类的可以使用TPM数据。</p>
<h1 id="call-snp">call snp</h1>

      </main>

      <div id="fastSearch">
   <input id="searchInput" tabindex="0">
   <ul id="searchResults">
   </ul>
</div>
<footer>
  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.2/highlight.min.js"></script>
  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.2/languages/r.min.js"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.1.2/languages/python.min.js"></script>
  <script src ="/js/search.js"></script>
  <script src="../../js/fuse.js"></script> 
 <script src="../../js/fastsearch.js"></script>

  <script>
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  </script>
  
  <script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script>
  <span id="busuanzi_container_site_pv">本站总访问量<span id       ="busuanzi_value_site_pv"></span>次</span>
  <br>
  <span >
    <mark>CMD+/</mark>对网站进行搜索
  </span>
  
  
<script async src="../../js/center-image.js"></script>
<script type="text/javascript" id="MathJax-script" async
  src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js">
</script>
<center>
Liripo <SPAN id=span_dt_dt style="color: #0196e3;"></SPAN>
<SCRIPT language=javascript>
function show_date_time(){
window.setTimeout("show_date_time()", 1000);
BirthDay=new Date("7/8/2019 00:00:00");
today=new Date();
timeold=(today.getTime()-BirthDay.getTime());
sectimeold=timeold/1000
secondsold=Math.floor(sectimeold);
msPerDay=24*60*60*1000
e_daysold=timeold/msPerDay
daysold=Math.floor(e_daysold);
e_hrsold=(e_daysold-daysold)*24;
hrsold=Math.floor(e_hrsold);
e_minsold=(e_hrsold-hrsold)*60;
minsold=Math.floor((e_hrsold-hrsold)*60);
seconds=Math.floor((e_minsold-minsold)*60);
span_dt_dt.innerHTML=""+daysold+"天"+hrsold+"小时"+minsold+"分"+seconds+"秒";
}
show_date_time();
</SCRIPT>
</center>
  
</footer>


    </div>
  </body>
</html>
